File: Programmers JS/프린터.js
function solution(priorities, location) {
  var answer = 0;

  let queue = priorities.map((prior, idx) => {
    return [prior, idx];
  });

  let pop;
  while (queue.length > 0) {
    pop = queue.shift();
    let Max = pop[0];

    for (let i = 0; i < queue.length; i++) {
      if (Max < queue[i][0]) {
        Max = queue[i][0];
        break;
      }
    }
    if (Max !== pop[0]) {
      queue.push(pop);
    } else {
      answer++;
      if (pop[1] === location) {
        break;
      }
    }
  }
  return answer;
}

File: Programmers/프린터.py
from collections import deque
def solution(priorities, location):
    answer = 0
    document = []
    for i in range(len(priorities)):
        document.append(i) 
  
    queue = deque(priorities)
    doc_queue = deque(document)
    while queue:
        pop_prior = queue.popleft()
        pop_doc = doc_queue.popleft()
        
        if len(queue) > 0:
            if max(queue) > pop_prior:
                queue.append(pop_prior)
                doc_queue.append(pop_doc)
            else:
                answer += 1
                if pop_doc == location:
                    break;
        else:
            answer += 1

    return answer